CRM/Contribute/BAO/Contribution.php

```diff
@@ -4577,11 +4570,13 @@ public static function completeOrder($input, &$ids, $objects, $transaction = NUL
       CRM_Activity_BAO_Activity::addActivity($contribution, NULL, $targetContactID);
     }
 
+    $isEmailReceipt = !array_key_exists('is_email_receipt', $values) || $values['is_email_receipt'] == 1;
+    if (isset($input['is_email_receipt'])) {
+      $isEmailReceipt = $input['is_email_receipt'];
+    }
```

Comment:
@artfulrobot yes - probably - I was focussed on moving existing lines to a new order so we don't do

- set this
- set it to something else possibly
- set it again in case we changed it & didn't mean to

& instead do

- set it to something else possibly 
- set it to $input['is_email_receipt'] if that has been passed in as it clobbers

Reply:
ie. $input always overrides $values - if set.